engine/security/rootless.md

```diff
@@ -0,0 +1,276 @@
+---
+
+Rootless mode allows running the Docker daemon and containers as a non-root
+user, for the sake of mitigating potentail vulnerabilities in the daemon and
+the container runtime.
+
+Rootless mode does not require the root privileges even for installation of the
+Docker daemon, as long as [the prerequisites](prerequiresites) are satisfied.
+
+Rootless mode was introduced in Docker Engine 19.03.
+
+> **Note**:
+> Rootless mode is an experimental feature and has [limitations](known-limitations).
+
+## How it works
+Rootless mode executes the Docker daemon and containers inside a user namespace.
+This is very similar to [`userns-remap` mode](userns-remap.md), except that
+`userns-remap` mode executes the daemon as the root.
+
+Rootless mode does not use binaries with SETUID bits or file capabilities,
+except `newuidmap` and `newgidmap`, which are needed for allowing multiple
+UIDs/GIDs to be used in the user namespace.
+
+## Prerequisites
+
+- `newuidmap` and `newgidmap` need to be installed on the host. These commands
+  are provided by the `uidmap` package on most distros.
+
+- `/etc/subuid` and `/etc/subgid` should contain at least 65,536 subordinate
+  UIDs/GIDs for the user. In the following example, the user `testuser` has
+  65,536 subordinate UIDs/GIDs (231072-296607).
+
+```console
+$ id -u
+1001
+$ whoami
+testuser
+$ grep ^$(whoami): /etc/subuid
+testuser:231072:65536
+$ grep ^$(whoami): /etc/subgid
+testuser::231072:65536
+```
+
+### Distribution-specific hint
+
+Using Ubuntu kernel is recommended.
+
+#### Ubuntu
+- No preparation is needed.
+
+- `overlay2` storage driver  is enabled by default
+  ([Ubuntu-specific kernel patch](https://kernel.ubuntu.com/git/ubuntu/ubuntu-bionic.git/commit/fs/overlayfs?id=3b7da90f28fe1ed4b79ef2d994c81efbc58f1144)).
+
+- Known to work on Ubuntu 16.04 and 18.04.
+
+#### Debian GNU/Linux
+- Add `kernel.unprivileged_userns_clone=1` to `/etc/sysctl.conf` (or
+  `/etc/sysctl.d`) and run `sudo sysctl --system`.
+
+- To use `overlay2` storage driver (recommended), run
+  `sudo modprobe overlay permit_mounts_in_userns=1`
+   ([Debian-specific kernel patch, introduced in Debian 10](https://salsa.debian.org/kernel-team/linux/blob/283390e7feb21b47779b48e0c8eb0cc409d2c815/debian/patches/debian/overlayfs-permit-mounts-in-userns.patch)).
+   Put the configuration to `/etc/modprobe.d` for persistence.
+
+- Known to work on Debian 9 and 10. 
+  `overlay2` is only supported since Debian 10 and needs `modprobe`
+  configuration described above.
+
+#### Arch Linux
+- Add `kernel.unprivileged_userns_clone=1` to `/etc/sysctl.conf` (or
+  `/etc/sysctl.d`) and run `sudo sysctl --system`
+
+#### openSUSE
+- `sudo modprobe ip_tables iptable_mangle iptable_nat iptable_filter` is required.
+  This is might be required on other distros as well depending on the configuration.
+
+- Known to work on openSUSE 15.
+
+#### Fedora 31 and later
+- Run `sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"`
```

Comment:
Should we explain what this does? Something like;

```suggestion
- Fedora 31 uses cgroups v2 by default, which is not yet supported by the containerd runtime.
  Run `sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"`
  to use cgroups v1.
```

(open to suggestions there)